metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -692,6 +702,136 @@ def get_lineage_if_enabled(
+                column_profile.uniqueProportion = float(
+                    column_params[self.source_config.profiling.unique_proportion]
+                )
+            if self.source_config.profiling.null_count in column_params:
+                column_profile.nullCount = int(
+                    column_params[self.source_config.profiling.null_count]
+                )
+            if self.source_config.profiling.null_proportion in column_params:
+                column_profile.nullProportion = float(
+                    column_params[self.source_config.profiling.null_proportion]
+                )
+            if self.source_config.profiling.min in column_params:
+                column_profile.min = column_params[self.source_config.profiling.min]
+            if self.source_config.profiling.max in column_params:
+                column_profile.max = column_params[self.source_config.profiling.max]
+            if self.source_config.profiling.mean in column_params:
+                column_profile.mean = column_params[self.source_config.profiling.mean]
+            if self.source_config.profiling.median in column_params:
+                column_profile.median = column_params[
+                    self.source_config.profiling.median
+                ]
+            if self.source_config.profiling.stdev in column_params:
+                column_profile.stdev = column_params[self.source_config.profiling.stdev]
+
+            dataset_profile.fieldProfiles.append(column_profile)
+
+        if partition_spec:
+            # inject partition level stats
+            dataset_profile.partitionSpec = PartitionSpecClass(
+                partition=partition_spec,
+                type=PartitionTypeClass.PARTITION,
+            )
+
+        mcp = MetadataChangeProposalWrapper(
+            entityType="dataset",
+            entityUrn=mce.proposedSnapshot.urn,
+            changeType=ChangeTypeClass.UPSERT,
+            aspectName="datasetProfile",
+            aspect=dataset_profile,
+        )
+        return mcp
+
+    def get_profile_if_enabled(
+        self, mce: MetadataChangeEventClass, database_name: str, table_name: str
+    ) -> List[Optional[MetadataChangeProposalWrapper]]:
+        if self.source_config.profiling.enabled:
+            if self.source_config.profiling.partitioned:
+                # for cross-account ingestion
+                if self.source_config.catalog_id:
+                    response = self.glue_client.get_partitions(
+                        DatabaseName=database_name,
+                        Name=table_name,
+                        CatalogId=self.source_config.catalog_id,
+                    )
+                else:
+                    response = self.glue_client.get_partitions(
+                        DatabaseName=database_name, Name=table_name
+                    )
+
+                partitions = response["partitions"]
+
+                mcps = []
+                for p in partitions:
```

Comment:
I think it would make sense to be able to specify the partition itself as well otherwise it will always profile all the partitions.
Something like this.
`partition_key: updated_at`
`partition_value: 2022-01-01`

Reply:
Do you mean like an `AllowDenyPattern` for partitions?